src/Modules/Visualization/ShowField.cc

```diff
@@ -448,754 +538,293 @@ void GeometryBuilder::renderFacesLinear(
-        fld->get_value(tvals[0], cells[0]);
+          if (fld->is_scalar())
+          {
+            fld->get_value(svals[0], *fiter);
+            textureCoords[0].x = coordinateMap->valueToColor(svals[0]).r();
+          }
+          else if (fld->is_vector())
+          {
+            fld->get_value(vvals[0], *fiter);
+            textureCoords[0].x = coordinateMap->valueToColor(vvals[0]).r();
+          }
+          else if (fld->is_tensor())
+          {
+            fld->get_value(tvals[0], *fiter);
+            textureCoords[0].x = coordinateMap->valueToColor(tvals[0]).r();
+          }
 
-        if (cells.size() > 1)
-        {
-          fld->get_value(tvals[1], cells[1]);
+          for (size_t i = 0; i < numNodes; ++i)
+            textureCoords[i].y = textureCoords[i].x = textureCoords[0].x;
         }
-        else
+        // Data at nodes
+        else if (fld->basis_order() == 1)
         {
-          svals[1] = svals[0];
+          if (fld->is_scalar())
+          {
+            for (size_t i = 0; i<numNodes; ++i)
+            {
+              fld->get_value(svals[i], nodes[i]);
+              textureCoords[i].x = textureCoords[i].y = coordinateMap->valueToColor(svals[i]).r();
+            }
+          }
+          else if (fld->is_vector())
+          {
+            for (size_t i = 0; i<numNodes; ++i)
+            {
+              fld->get_value(vvals[i], nodes[i]);
+              textureCoords[i].x = textureCoords[i].y = coordinateMap->valueToColor(vvals[i]).r();
+            }
+          }
+          else if (fld->is_tensor())
+          {
+            for (size_t i = 0; i<numNodes; ++i)
+            {
+              fld->get_value(tvals[i], nodes[i]);
+              textureCoords[i].x = textureCoords[i].y = coordinateMap->valueToColor(tvals[i]).r();
+            }
+          }
         }
-
-        face_colors[0] = map->valueToColor(tvals[0]);
-        face_colors[1] = map->valueToColor(tvals[1]);
       }
 
-      state.set(RenderState::IS_DOUBLE_SIDED, true);
-
-      addFaceGeom(points, normals, withNormals, iboIndex, iboBuffer, vboBuffer,
-        colorScheme, face_colors, state);
-    }
-
-    // Element data (faces)
-    else if (fld->basis_order() == 0 && mesh->dimensionality() == 2)
-    {
-      auto map = colorMap.get();
-      //one possible color, each node that color.
-      svals.resize(1);
-      vvals.resize(1);
-      tvals.resize(1);
-      face_colors.resize(nodes.size());
-      if (fld->is_scalar())
+      switch(writeCase)
       {
-        fld->get_value(svals[0], *fiter);
-        face_colors[0] = map->valueToColor(svals[0]);
-      }
-      else if (fld->is_vector())
-      {
-        fld->get_value(vvals[0], *fiter);
-        face_colors[0] = map->valueToColor(vvals[0]);
-      }
-      else if (fld->is_tensor())
-      {
-        fld->get_value(tvals[0], *fiter);
-        face_colors[0] = map->valueToColor(tvals[0]);
+        case 0b000: writeTri(vboBuffer, iboBuffer, iboIndex, points); break;
```

Comment:
Please use `enum` labels for these, we're not writing assembly code here

Reply:
If I had it my way...

Reply:
You can put all the assembly you want in the shader files, i ignore those...